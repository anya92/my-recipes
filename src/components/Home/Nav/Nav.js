import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const StyledNav = styled.div`
  display: flex;
  max-width: 900px;
  margin: 20px auto;
  font-size: 1.2rem;
  div, a {
    width: calc(100% / 3);
    text-align: center;
    cursor: pointer;
    user-select: none;
  }
  a {
    color: #333;
    text-decoration: none;
    font-weight: 700;
  }
`;

const Nav = ({ index, handleNext, handlePrevious, recipesLength }) => (
  <StyledNav>
    {
      index > 0
      ? <div onClick={handlePrevious} id="previous">previous</div>
      : <div />
    }
    <Link to="/recipes">view all recipes</Link>
    {
      index < recipesLength - 3
      ? <div onClick={handleNext} id="next">next</div>
      : <div />
    }
  </StyledNav>
);

Nav.propTypes = {
  index: PropTypes.number.isRequired,
  handleNext: PropTypes.func.isRequired,
  handlePrevious: PropTypes.func.isRequired,
  recipesLength: PropTypes.number.isRequired,
};

export default Nav;
