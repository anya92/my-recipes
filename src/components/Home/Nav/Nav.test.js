import React from 'react';
import { shallow } from 'enzyme';

import Nav from './Nav';

let index;
let recipesLength;
let handleNext;
let handlePrevious;
let wrapper;

beforeEach(() => {
  index = 3;
  recipesLength = 25;
  handleNext = jest.fn();
  handlePrevious = jest.fn();
  wrapper = shallow(<Nav
    index={index}
    recipesLength={recipesLength}
    handleNext={handleNext}
    handlePrevious={handlePrevious}
  />);
});

test('should not render previous if index = 0', () => {
  index = 0;
  wrapper = shallow(<Nav
    index={index}
    recipesLength={recipesLength}
    handleNext={handleNext}
    handlePrevious={handlePrevious}
  />);
  expect(wrapper).toMatchSnapshot();
});

test('should render previous and next if index > 0', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should not render next if index > length of all recipes - 3', () => {
  index = 23;
  wrapper = shallow(<Nav
    index={index}
    recipesLength={recipesLength}
    handleNext={handleNext}
    handlePrevious={handlePrevious}
  />);
  expect(wrapper).toMatchSnapshot();
});

test('should called handleNext when next is clicked', () => {
  wrapper.find('#next').simulate('click');
  expect(handleNext).toHaveBeenCalled();
});

test('should called handlePrevious when previous is clicked', () => {
  wrapper.find('#previous').simulate('click');
  expect(handlePrevious).toHaveBeenCalled();
});
