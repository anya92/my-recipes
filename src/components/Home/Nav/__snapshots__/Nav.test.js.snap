// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render next if index > length of all recipes - 3 1`] = `
<styled.div>
  <div
    id="previous"
    onClick={[Function]}
  >
    previous
  </div>
  <Link
    replace={false}
    to="/recipes"
  >
    view all recipes
  </Link>
  <div />
</styled.div>
`;

exports[`should not render previous if index = 0 1`] = `
<styled.div>
  <div />
  <Link
    replace={false}
    to="/recipes"
  >
    view all recipes
  </Link>
  <div
    id="next"
    onClick={[Function]}
  >
    next
  </div>
</styled.div>
`;

exports[`should render previous and next if index > 0 1`] = `
<styled.div>
  <div
    id="previous"
    onClick={[Function]}
  >
    previous
  </div>
  <Link
    replace={false}
    to="/recipes"
  >
    view all recipes
  </Link>
  <div
    id="next"
    onClick={[Function]}
  >
    next
  </div>
</styled.div>
`;
